lib/webidl2.js

```diff
@@ -1039,44 +1039,57 @@
     }
 
     function enum_() {
-      if (!consume("enum")) return;
+      const base = consume("enum");
+      if (!base) return;
+      const trivia = { base: base.trivia };
       const name = consume(ID) || error("No name for enum");
+      trivia.name = name.trivia;
       const vals = [];
       const ret = current = {
         type: "enum",
         name: sanitize_name(name.value, "enum"),
-        values: vals
+        values: vals,
+        trivia
       };
-      consume("{") || error("No curly for enum");
+      const open = consume("{") || error("No curly for enum");
+      trivia.open = open.trivia;
       let value_expected = true;
       while (true) {
-        if (consume("}")) {
+        const close = consume("}");
```

Comment:
Had expected to see “No closing } for enum” somewhere as an error. Is that possible?

Reply:
Currently `enum X {` throws with *Unexpected value in enum*, which is weird here. I introduced `eof` type in #185, maybe we can do better with it by `Unexpected end of file`.

Reply:
Filed #208, we may revisit it later.